valora-inc/wallet
# chore: ensure backwards compatibility when dapps v2 data is used

### Description

We persist the dapps data in redux, so existing users will have the v1 dapps data schema. When the experiment is switched on for the user, they will first land on the new dapps screens with v1 data while the request to fetch new data is kicked off. During this time, the screen tries to render and crashes due to data incompatibility. 

### Test plan

To repro:
1. Load the app, visit the dapps screen. Observe current dapps screen loads.
2. In statsig, assign the search or search + filter variant to your user
3. Reload the app, visit the dapps screen. Note that the app crashes, and note that the v2 data is not received in time (so it is still v1 data in the redux store). **This means that the user will continue to to get app crashes every time they visit the dapps screen.**

https://user-images.githubusercontent.com/20150449/233394570-3fe528a6-907a-4239-9b7c-2178cf2299ce.mp4



### Related issues

N/A

### Backwards compatibility

Y
